File: internal/installer/steps/run_onboard.go
package steps

import (
	"context"

	"github.com/SolsticeSauer/solsclaw_beta/internal/installer"
)

type RunOnboard struct{}

func (RunOnboard) ID() string                              { return "onboard-daemon" }
func (RunOnboard) Label() string                           { return "Register OpenClaw daemon" }
func (RunOnboard) ShouldRun(_ installer.StepContext) bool  { return true }

func (s RunOnboard) Run(ctx context.Context, sc installer.StepContext) error {
	return streamCommand(ctx, sc, s.ID(), "openclaw", "onboard", "--install-daemon", "--non-interactive")
}

File: packages/installer-server/src/lib/steps/runOnboard.ts
import { execa } from 'execa';
import type { InstallerStep } from './index.js';

export const runOnboardStep: InstallerStep = {
  id: 'onboard-daemon',
  label: 'Register OpenClaw daemon (launchd / systemd / Windows service)',
  shouldRun: () => true,
  async run({ bus }) {
    const sub = execa('openclaw', ['onboard', '--install-daemon', '--non-interactive'], { all: true });
    sub.all?.on('data', (chunk: Buffer) => {
      const text = chunk.toString().trimEnd();
      if (text) bus.log(runOnboardStep.id, 'info', text);
    });
    await sub;
  },
};
